### PR TITLE
Add notify-path API to templatized template. Fixes #2505.

### LINKED ISSUE #2505: Templatizer: Changes to array/object properties from the parent scope not notified
When the templatized template is not the templatizer client, changes to array/object properties from the parent scope are not notified to template instances.

Originally filed as https://github.com/PolymerElements/iron-list/issues/109

Root caused to https://github.com/Polymer/polymer/blob/master/src/lib/template/templatizer.html#L243 not adding notifyPath API, or any related path handling capability to the template.

### PR DESCRIPTION
**Pull request by kevinpschaaf**

### CODE REVIEW

**Comment by samccone** [comment]
Is there a test we can put into place for this¿

File: src/lib/template/templatizer.html [hunk 1]
```diff
@@ -311,6 +315,15 @@
       Polymer.Base._pathEffector.apply(this, arguments);
     },
 
+    // Overrides Base notify-path module (for non-PE templates)
+    _pathEffectorTemplate: function(path, value, fromAbove) {
+      if (this._forwardParentPath) {
+        if (path.indexOf(this._parentPropPrefix) === 0) {
+          this._forwardParentPath(path.substring(8), value);
```
Comment (sorvell) [inline src/lib/template/templatizer.html]: 8? need comment.

File: src/standard/notify-path.html [hunk 2]
```diff
@@ -483,6 +483,27 @@
           this._notifySplice(array, path, 0, args.length, []);
         }
         return ret;
+      },
+
+      // TODO(kschaaf): This is the path analogue to Polymer.Bind.prepareModel,
```
Comment (sorvell) [inline src/standard/notify-path.html]: Let's make an issue for this and link it here.

File: src/lib/template/templatizer.html (hunk 1 above)
Comment (sorvell) [inline src/lib/template/templatizer.html]: `this._parentPropPrefix.length`

File: src/lib/template/templatizer.html [hunk 3]
```diff
@@ -311,6 +315,15 @@
       Polymer.Base._pathEffector.apply(this, arguments);
     },
 
+    // Overrides Base notify-path module (for non-PE templates)
+    _pathEffectorTemplate: function(path, value, fromAbove) {
```
Comment (sorvell) [inline src/lib/template/templatizer.html]: Confirm why this is needed.

File: src/lib/template/templatizer.html [hunk 4]
```diff
@@ -311,6 +315,15 @@
...
+      if (this._forwardParentPath) {
```
Comment (sorvell) [inline src/lib/template/templatizer.html]: DRY issue with `_pathEffector`

File: src/lib/template/templatizer.html [hunk 5]
```diff
@@ -241,10 +243,12 @@
         // Instance setup
         if (template != this) {
           Polymer.Bind.prepareInstance(template);
-          template._forwardParentProp =
-            this._forwardParentProp.bind(this);
+          template._forwardParentProp = this._forwardParentProp.bind(this);
         }
         this._extendTemplate(template, proto);
+        if (template != this) {
+          template._pathEffector = this._pathEffectorTemplate.bind(this);
```
Comment (sorvell) [inline src/lib/template/templatizer.html]: Consider instead unconditionally doing:

`template._pathEffector = this._pathEffector.bind(template, this);`

**Comment by sorvell** [comment]
LGTM

**Pull request merged**